portfolio/src/main/webapp/script.js

```diff
@@ -216,3 +195,103 @@ function initializeAuth() {
+  controlUI.classList.add("tropicalStormInfoControl");
+  controlDiv.appendChild(controlUI);
+
+  let controlText = document.createElement("div");
+  controlText.textContent =
+    "Hover over a path to get info on the tropical storm.";
+  controlUI.appendChild(controlText);
+}
+
+/**
+ * Generates a map from Google Map API
+ */
+function initializeMap() {
+  fetch("/tropical-storm-data")
+    .then((response) => response.json())
+    .then((tropicalStormData) => {
+      map = new google.maps.Map(document.getElementById("map"), {
+        center: { lat: 35.5733225, lng: -80.7187387 },
+        zoom: 8,
+        disableDefaultUI: true,
+        styles: [
+          {
+            featureType: "poi",
+            stylers: [{ visibility: "off" }]
+          },
+          {
+            featureType: "transit",
+            stylers: [{ visibility: "off" }]
+          },
+          {
+            featureType: "road",
+            stylers: [{ visibility: "off" }]
+          }
+        ]
+      });
+
+      let infoControlDiv = document.createElement("div");
+      let infoControl = new InfoControl(infoControlDiv, map);
+
+      infoControlDiv.index = 1;
+      map.controls[google.maps.ControlPosition.TOP_CENTER].push(infoControlDiv);
+
+      Object.keys(tropicalStormData).forEach((tropicalStormID) => {
+        let tropicalStormPathCoords = [];
+
+        tropicalStormData[tropicalStormID].forEach((dataPoint) => {
+          tropicalStormPathCoords.push({
+            lat: dataPoint["lat"],
+            lng: dataPoint["lon"]
+          });
+        });
+
+        let randomHue = Math.floor(Math.random() * 360) + 90;
+        let randomColor = `hsl(${randomHue}, 100%, 40%)`;
+
+        tropicalStormPath = new google.maps.Polyline({
+          path: tropicalStormPathCoords,
+          geodesic: true,
+          strokeColor: randomColor,
+          strokeOpacity: 0.75,
+          strokeWeight: 8,
+          map: map
+        });
+
+        /*
+         Event handling to "highlight" a hurricane path and displaying its info
+         */
+        google.maps.event.addListener(
+          tropicalStormPath,
+          "mouseover",
+          function() {
+            infoControlDiv.firstChild.textContent = `${tropicalStormData[tropicalStormID][0]["name"]} ${tropicalStormData[tropicalStormID][0]["season"]} | First Landfall on ${tropicalStormData[tropicalStormID][0]["date"]} | ID:${tropicalStormID}`;
+            this.setOptions({
+              strokeOpacity: 1,
+              strokeWeight: 10
+            });
+          }
+        );
+        google.maps.event.addListener(
+          tropicalStormPath,
+          "mouseout",
+          function() {
+            infoControlDiv.firstChild.textContent =
+              "Hover over a path to get info on the tropical storm.";
+            this.setOptions({
+              strokeOpacity: 0.75,
+              strokeWeight: 8
+            });
+          }
+        );
+      });
```

Comment:
This function lambda body is pretty huge. Could you break this up into a few separate functions?